src/lib/plugin.js

```diff
@@ -59,15 +60,14 @@ function * resolveWebfingerOptions (identifier) {
   return newOptions
 }
 
-function * requestRetry (opts, errorMessage, credentials) {
+function * requestRetry (requestOptions, retryOptions) {
   let delay = backoffMin
+  const start = Date.now()
+  const timeout = retryOptions.timeout
   while (true) {
-    debug('connecting to account ' + opts.uri)
+    debug('connecting to account ' + requestOptions.uri)
     try {
-      let res = yield request(Object.assign(
-        {json: true},
-        requestCredentials(credentials),
-        opts))
+      const res = yield request(requestOptions)
       if (res.statusCode >= 400 && res.statusCode < 500) {
         break
       }
```

Comment:
While you're changing this stuff could you also change the fact that it will hit this debug line even if it doesn't need to retry the connection? `debug('request status ' + res.statusCode + ' retrying connection')`